src/App.tsx

```diff
@@ -1,26 +1,186 @@
+      setError(ErrorType.UnableToLoad);
+    }
+  };
+
+  useEffect(() => {
+    downloadTodos();
+    inputRef.current?.focus();
+  }, []);
+
+  useEffect(() => {
+    if (todos) {
+      switch (query) {
+        case FilterType.Active:
+          setFilteredTodos(todos.filter(todo => !todo.completed));
+          break;
+        case FilterType.Complited:
+          setFilteredTodos(todos.filter(todo => todo.completed));
+          break;
+        default:
+          setFilteredTodos(todos);
+          break;
+      }
+    }
+  }, [todos, query]);
+
+  const returnError = () => {
+    switch (error) {
+      case ErrorType.EmptyTitle:
+        return 'Title should not be empty';
+      case ErrorType.UnableToDelete:
+        return 'Unable to delete a todo';
+      case ErrorType.UnableToLoad:
+        return 'Unable to load todos';
+      case ErrorType.UnableToUpdate:
+        return 'Unable to update a todo';
+      case ErrorType.UnableToAdd:
+        return 'Unable to add a todo';
+      default:
+        return null;
+    }
+  };
+
+  useEffect(() => {
+    const timeout = setTimeout(() => {
+      setError(null);
+    }, 3000);
+
+    return () => {
+      clearTimeout(timeout);
+    };
+  }, [error]);
+
+  const returnLeftNumber = () => {
+    return todos.filter(todo => todo.completed != true).length;
+  };
+
+  const handleSetQuery = (passedQuery: FilterType) => {
+    setQuery(passedQuery);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+    if (inputRef.current?.value.trim()) {
+      inputRef.current.disabled = true;
+      setTmpTodo({
+        title: inputRef.current.value.trim(),
+        id: -1,
+        completed: false,
+      });
+
+      postTodo(inputRef.current.value.trim())
+        .then(response => {
+          setTodos(prev => [...prev, response]);
+
+          if (inputRef.current) {
+            inputRef.current.value = '';
+          }
+        })
+        .catch(() => setError(ErrorType.UnableToAdd))
+        .finally(() => {
+          if (inputRef.current) {
+            inputRef.current.disabled = false;
+            inputRef.current.focus();
+          }
+
+          setTmpTodo(null);
+        });
+    } else {
+      setError(ErrorType.EmptyTitle);
+    }
+  };
+
+  const removeOneTodo = (id: number) => {
+    setUpdatedTodosId(prev => [...prev, id]);
+
+    removeTodo(id)
+      .then(() => {
+        setTodos(prev => prev.filter(el => el.id != id));
+      })
+      .catch(() => setError(ErrorType.UnableToDelete))
+      .finally(() => {
+        setUpdatedTodosId(prev => prev.filter(el => el != id));
+        inputRef.current?.focus();
+      });
+  };
+
+  const removeAllComplited = () => {
+    const tmp = todos.filter(todo => todo.completed).map(todo => todo.id);
+
+    setUpdatedTodosId(tmp);
+
+    tmp.map(id => {
+      removeOneTodo(id);
+    });
+  };
+
+  const isClearAllCompletedActive = () => {
+    return !todos.some(todo => todo.completed);
+  };
```

Comment:
This names makes me think that it keeps value of type `boolean`. Change the name or remove this function and compute to `boolean` value here